compiler/java/src/test/java/com/c/lexer/LexerTest.java

```diff
@@ -0,0 +1,540 @@
+
+        String arbitraryDummy2 = "\n" +
+                "    }\n" +
+                "\n" +
+                "    private void dummy2() {\n" +
+                "        ";
+        checkExpectToken(lexer, new Token(TokenType.ARBITRARY, arbitraryDummy2, 21, 13));
+
+        // Start CCom section (now with block comments)
+        checkExpectToken(lexer, new Token(TokenType.COMMENT_BLOCK_OPEN_IDENTIFIER, 25, 9));
+
+        checkExpectToken(lexer, new Token(TokenType.STRING, "a", 26, 9));
+        checkExpectToken(lexer, new Token(TokenType.BRACE_OPEN, 26, 13));
+
+        String payloadDummy2 = "StringBuilder str = new StringBuilder(\"/* { } \");\n" + // "//" replace by ""
+                "        ";
+        checkExpectToken(lexer, new Token(TokenType.ARBITRARY, payloadDummy2, 27, 9));
+
+        // Close conditional comment
+        checkExpectToken(lexer, new Token(TokenType.BRACE_CLOSE, 28, 9));
+        checkExpectToken(lexer, new Token(TokenType.COMMENT_BLOCK_CLOSE_IDENTIFIER, 28, 10));
+
+        // Arbitrary between sections
+        String arbitraryBetweenSection = "\n" +
+                "\n" +
+                "        ";
+        checkExpectToken(lexer, new Token(TokenType.ARBITRARY, arbitraryBetweenSection, 28, 12));
+
+        // Start CCom section (now with block comments)
+        checkExpectToken(lexer, new Token(TokenType.COMMENT_BLOCK_OPEN_IDENTIFIER, 30, 9));
+
+        checkExpectToken(lexer, new Token(TokenType.STRING, "b", 30, 12));
+        checkExpectToken(lexer, new Token(TokenType.NOT_EQUALS, 30, 16));
+        checkExpectToken(lexer, new Token(TokenType.LESS, 30, 19));
+        checkExpectToken(lexer, new Token(TokenType.IDENTIFIER, "testIdentifier", 30, 21));
+        checkExpectToken(lexer, new Token(TokenType.BRACE_OPEN, 30, 36));
+
+        // Payload
+        String payloadLast = "StringBuilder str2 = new StringBuilder(\"/* { } \");\n" + // "//" replaced by ""
+                "        int z = 42;\n" +
+                "           ";
+        checkExpectToken(lexer, new Token(TokenType.ARBITRARY, payloadLast, 31, 9));
+
+        // Close conditional comment
+        checkExpectToken(lexer, new Token(TokenType.BRACE_CLOSE, 33, 12));
+        checkExpectToken(lexer, new Token(TokenType.COMMENT_BLOCK_CLOSE_IDENTIFIER, 33, 13));
+
+        // "//" not replaced (!)
+        String arbitraryEnd = """
+
+                    }
+
+                    private String rest() {
+                        return "This is the rest // /* */ // { }";
+                    }
+
+                }
+                """;
+        checkExpectToken(lexer, new Token(TokenType.ARBITRARY, arbitraryEnd, 33, 15));
+    }
+
+
+    // ---------------------------------------------- Exceptions -------------------------------------------------------
+    // TODO
+
+
+    // ------------------ Single statements (file only consists of a conditional comments section ----------------------
```

Comment:
FYI: A single statement is, like its name suggests, a single statement list. This is not to confuse with a single conditional comments section. A single statement also has another entrypoint grammar node. Instead of `CONTENT` in enters the grammar at `STMT_LST`.

Reply:
Thanks for the clarification, I misconfused that.